src/main/java/frc/robot/subsystems/swerve/Swerve.java

```diff
@@ -332,22 +332,39 @@ protected void rotateToAngle(MirrorableRotation2d targetAngle) {
     // todo - maybe move some of work to SwerveMath class
     private Rotation2d calculateProfiledAngleSpeedToTargetAngle(MirrorableRotation2d targetAngle) {
         Rotation2d currentAngle = POSE_ESTIMATOR.getCurrentPose().getRotation();
-        return Rotation2d.fromDegrees(SwerveConstants.PROFILED_ROTATION_PID_DEGREES_CONTROLLER.calculate(
+        return Rotation2d.fromDegrees(SwerveConstants.ROTATION_PID_DEGREES_CONTROLLER.calculate(
                 currentAngle.getDegrees(),
                 targetAngle.get().getDegrees()
         ));
     }
 
     // todo - maybe move some of work to SwerveMath class
-    private Rotation2d getAimAssistThetaVelocity(AimAssist aimAssist) {
+    private ChassisSpeeds applyAimAssistRotationVelocity(ChassisSpeeds currentSpeeds) {
         if (currentState.getAimAssist().equals(AimAssist.NONE)) {
-            return new Rotation2d();
+            return currentSpeeds;
         }
-        Rotation2d pidVelocity = calculateProfiledAngleSpeedToTargetAngle(aimAssist.targetAngleSupplier.get());
-        //todo - make value have same range like joystick
+        //PID
+        Rotation2d pidVelocity = calculateProfiledAngleSpeedToTargetAngle(currentState.getAimAssist().targetAngleSupplier.get());
+
+        //Magnitude Factor
+        double driveMagnitude = // todo - move to math
+                Math.sqrt((currentSpeeds.vyMetersPerSecond * currentSpeeds.vyMetersPerSecond) + (currentSpeeds.vxMetersPerSecond * currentSpeeds.vxMetersPerSecond));
+        double angularVelocityRads =
+                pidVelocity.getRadians() * SwerveConstants.AIM_ASSIST_MAGNITUDE_FACTOR / (driveMagnitude + SwerveConstants.AIM_ASSIST_MAGNITUDE_FACTOR);
+
+        //Joystick Output
+        double angularVelocityWithJoystick = angularVelocityRads + currentSpeeds.omegaRadiansPerSecond;
+
+        //Clamp
+        double clampedAngularVelocity = MathUtil.clamp(
+                angularVelocityWithJoystick,
+                -SwerveConstants.MAX_ROTATIONAL_SPEED_PER_SECOND.getRadians(),
+                SwerveConstants.MAX_ROTATIONAL_SPEED_PER_SECOND.getRadians()
+        );
```

Comment:
consider making a clamp function that only gets two arguments and uses the second argument's minus to clamp.

Reply:
I don't want to create GBMath just for this. maybe in the future if there will be more functions to put there